test/integration/acceptance/get_asset_info_test.cpp

```diff
@@ -0,0 +1,145 @@
+   * @param asset is asset to query
+   * @return built query
+   */
+  auto makeQuery(const std::string &asset) {
+    return complete(baseQry().getAssetInfo(asset));
+  }
+
+  /**
+   * Creates valid GetAssetInfo query of the current asset
+   * @return built query
+   */
+  auto makeQuery() {
+    return makeQuery(kAssetId);
+  }
+
+  /**
+   * @param asset is asset name for checking
+   * @param domain is domain for checking
+   * @param role is role for checking
+   * @return a lambda that verifies that query response contains specified asset
+   */
+  auto checkValidAsset(const std::string &asset,
+                       const std::string &domain,
+                       uint8_t precision) {
+    return [&](const proto::QueryResponse &response) {
+      ASSERT_NO_THROW({
+        const auto &resp = boost::apply_visitor(
+            framework::SpecifiedVisitor<interface::AssetResponse>(),
+            response.get());
+        ASSERT_EQ(resp.asset().assetId(), asset);
+        ASSERT_EQ(resp.asset().domainId(), domain);
+        ASSERT_EQ(resp.asset().precision(), precision);
+      }) << "Actual response: "
+         << response.toString();
+    };
+  }
+
+  /**
+   * @return a lambda that verifies that query response contains specified asset
+   */
+  auto checkValidAsset() {
+    return checkValidAsset(kAssetId, kDomain, 1);
+  }
+
+  const std::string kNewRole = "rl";
+  IntegrationTestFramework itf;
+};
+
+/**
+ * C363 Get asset info with CanReadAssets permission
+ * @given a user with GetMyAccount permission
+ * @when GetAssetInfo is queried on the user
+ * @then there is a valid AccountResponse
+ * TODO: enable after IR-1668
+ */
+TEST_F(GetAssetInfo, DISABLED_Basic) {
+  prepareState().sendQuery(makeQuery(), checkValidAsset());
+}
+
+/**
+ * C365 Pass an empty asset id
+ * @given a user with all required permissions
+ * @when GetAssetInfo is queried on the empty asset name
+ * @then query is stateless invalid response
+ */
+TEST_F(GetAssetInfo, EmptyAsset) {
+  prepareState().sendQuery(
+      makeQuery(""),
+      checkQueryErrorResponse<
+          shared_model::interface::StatelessFailedErrorResponse>());
+}
+
+/**
+ * C366 Pass a non-existing asset id
+ * @given a user with all required permissions
+ * @when GetAssetInfo is queried on the user
+ * @then query is stateful invalid response
+ */
+TEST_F(GetAssetInfo, NonexistentAsset) {
+  prepareState().sendQuery(
+      makeQuery("inexistent#" + kDomain),
+      checkQueryErrorResponse<shared_model::interface::NoAssetErrorResponse>());
+}
+
+/**
+ * C364 Get asset info without CanReadAssets permission
+ * @given a user without any query-related permission
+ * @when GetAssetInfo is queried on the user
+ * @then query is stateful invalid response
+ */
+TEST_F(GetAssetInfo, NoPermission) {
+  prepareState({}).sendQuery(
+      makeQuery(),
+      checkQueryErrorResponse<
+          shared_model::interface::StatefulFailedErrorResponse>());
+}
```

Comment:
Isn't it useful to have also test for sending `GetAssetInfo` from non existent user? I think, it's covered in other tests, but still

Reply:
The task has no such test. But indeed that might be useful so I'll add it